File: server/router/client/external_link_domain.go
package client

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ExternalLinkDomainRouter struct{}

func (s *ExternalLinkDomainRouter) InitExternalLinkDomainRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	extDomainRouter := Router.Group("extDomain").Use(middleware.OperationRecord())
	extDomainRouterWithoutRecord := Router.Group("extDomain")
	extDomainPublicRouter := PublicRouter.Group("extDomain")

	var extDomainApi = v1.ApiGroupApp.ClientApiGroup.ExternalLinkDomainApi
	{
		extDomainRouter.POST("createExternalLinkDomain", extDomainApi.CreateExternalLinkDomain)   // 创建域名
		extDomainRouter.DELETE("deleteExternalLinkDomain", extDomainApi.DeleteExternalLinkDomain) // 删除域名
		extDomainRouter.PUT("updateExternalLinkDomain", extDomainApi.UpdateExternalLinkDomain)    // 更新域名
		extDomainRouter.POST("setDefaultDomain", extDomainApi.SetDefaultDomain)                   // 设置默认域名
	}
	{
		extDomainRouterWithoutRecord.GET("getExternalLinkDomainList", extDomainApi.GetExternalLinkDomainList) // 获取域名列表
		extDomainRouterWithoutRecord.GET("listCloudFiles", extDomainApi.ListCloudFiles)                       // 列出云存储文件
		extDomainRouterWithoutRecord.GET("compareDirectories", extDomainApi.CompareDirectories)               // 多云目录比对
		extDomainRouter.POST("pingCloud", extDomainApi.PingCloud)                                              // 检测云存储连接
		extDomainRouter.POST("deleteCloudFiles", extDomainApi.DeleteCloudFiles)                               // 批量删除云存储文件
		extDomainRouter.POST("uploadCloudFile", extDomainApi.UploadCloudFile)                                 // 上传文件到云存储
		extDomainRouterWithoutRecord.GET("searchCloudFiles", extDomainApi.SearchCloudFiles)                     // 全局搜索云存储文件
		extDomainRouterWithoutRecord.GET("getFileDownloadURL", extDomainApi.GetFileDownloadURL)                 // 获取文件下载链接
		extDomainRouterWithoutRecord.GET("downloadCloudFolder", extDomainApi.DownloadCloudFolder)               // 打包下载目录
	}
	{
		extDomainPublicRouter.GET("getDefaultDomain", extDomainApi.GetDefaultDomain) // 获取默认域名（公开）
	}
}

File: web/src/api/client/externalLinkDomain.js
import service from '@/utils/request'

/**
 * 创建外部链接域名
 * @param {Object} data 域名信息
 * @returns {Promise}
 */
export const createExternalLinkDomain = (data) => {
  return service({
    url: '/extDomain/createExternalLinkDomain',
    method: 'post',
    data: data
  })
}

/**
 * 删除外部链接域名
 * @param {Object} data { id }
 * @returns {Promise}
 */
export const deleteExternalLinkDomain = (data) => {
  return service({
    url: '/extDomain/deleteExternalLinkDomain',
    method: 'delete',
    data: data
  })
}

/**
 * 更新外部链接域名
 * @param {Object} data 域名信息
 * @returns {Promise}
 */
export const updateExternalLinkDomain = (data) => {
  return service({
    url: '/extDomain/updateExternalLinkDomain',
    method: 'put',
    data: data
  })
}

/**
 * 获取外部链接域名列表
 * @param {Object} data 分页参数
 * @returns {Promise}
 */
export const getExternalLinkDomainList = (data) => {
  return service({
    url: '/extDomain/getExternalLinkDomainList',
    method: 'get',
    params: data
  })
}

/**
 * 设置默认域名
 * @param {Object} data { id }
 * @returns {Promise}
 */
export const setDefaultDomain = (data) => {
  return service({
    url: '/extDomain/setDefaultDomain',
    method: 'post',
    data: data
  })
}

/**
 * 获取默认域名（公开接口）
 * @returns {Promise}
 */
export const getDefaultDomain = () => {
  return service({
    url: '/extDomain/getDefaultDomain',
    method: 'get'
  })
}

/**
 * 检测云存储连接
 * @param {Object} data { id }
 * @returns {Promise}
 */
export const pingCloud = (data) => {
  return service({
    url: '/extDomain/pingCloud',
    method: 'post',
    data: data
  })
}

/**
 * 列出云存储文件
 * @param {Object} params 查询参数
 * @returns {Promise}
 */
export const listCloudFiles = (params) => {
  return service({
    url: '/extDomain/listCloudFiles',
    method: 'get',
    params: params
  })
}

/**
 * 多云目录比对
 * @param {Object} params 查询参数
 * @returns {Promise}
 */
export const compareDirectories = (params) => {
  return service({
    url: '/extDomain/compareDirectories',
    method: 'get',
    params: params
  })
}

/**
 * 批量删除云存储文件
 * @param {Object} data { id, keys: string[], password: string }
 * @returns {Promise}
 */
export const deleteCloudFiles = (data) => {
  return service({
    url: '/extDomain/deleteCloudFiles',
    method: 'post',
    data: data
  })
}

/**
 * 上传文件到云存储
 * @param {FormData} formData 含 file、folder、id
 * @returns {Promise}
 */
export const uploadCloudFile = (formData) => {
  return service({
    url: '/extDomain/uploadCloudFile',
    method: 'post',
    headers: { 'Content-Type': 'multipart/form-data' },
    data: formData
  })
}

/**
 * 全局搜索云存储文件
 * @param {Object} params { id, keyword, maxKeys }
 * @returns {Promise}
 */
export const searchCloudFiles = (params) => {
  return service({
    url: '/extDomain/searchCloudFiles',
    method: 'get',
    params: params
  })
}

/**
 * 获取文件下载链接
 * @param {Object} params { id, key }
 * @returns {Promise}
 */
export const getFileDownloadURL = (params) => {
  return service({
    url: '/extDomain/getFileDownloadURL',
    method: 'get',
    params: params
  })
}

/**
 * 获取打包下载目录的 URL（直接打开下载）
 * @param {Object} params { id, prefix }
 * @returns {string} 下载链接
 */
export const getCloudFolderDownloadURL = (params) => {
  const baseURL = service.defaults?.baseURL || ''
  const query = new URLSearchParams(params).toString()
  return `${baseURL}/extDomain/downloadCloudFolder?${query}`
}
